assets/js/blocks/cart-checkout/cart/edit.js

```diff
@@ -171,60 +172,25 @@ const BlockSettings = ( { attributes, setAttributes } ) => {
  * Component to handle edit mode of "Cart Block".
  */
 const CartEditor = ( { className, attributes, setAttributes } ) => {
-	return (
-		<div className={ className }>
-			<ViewSwitcher
-				label={ __( 'Edit', 'woo-gutenberg-products-block' ) }
-				views={ [
-					{
-						value: 'full',
-						name: __( 'Full Cart', 'woo-gutenberg-products-block' ),
-					},
-					{
-						value: 'empty',
-						name: __(
-							'Empty Cart',
-							'woo-gutenberg-products-block'
-						),
-					},
-				] }
-				defaultView={ 'full' }
-				render={ ( currentView ) => (
+	const defaultView = 'full';
+	const [ previousView, setPreviousView ] = useState( defaultView );
+
+	const viewRenderCallback = ( currentView ) => {
+		if ( previousView !== currentView ) {
+			recordEditorEvent( 'cart_view_toggle', {
```

Comment:
We could move this to ViewSwitcher, or better wrap in a TrackedViewSwitcher component so we don't need to repeat this pattern every time.